rcldotnet_common/DllLoadUtils.cs

```diff
@@ -243,7 +249,7 @@ public class DllLoadUtilsWindowsDesktop : DllLoadUtilsAbs
 
             public override IntPtr LoadLibrary(string fileName)
             {
-                string libraryName = fileName + "_native.dll";
+                string libraryName = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, AssemblyDirectory, fileName + "_native.dll");
```

Comment:
IMHO this is a bit of a patch. 
The search path should not be set to a hard path, as you cannot change the path of the library now. it is better to use the "AddDllDirectory" function. [https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-adddlldirectory](AddDllDirectory). 
This is also the recommended solution from microsoft in the LoadLibraryA function [https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibrarya](LoadLibraryA)

Reply:
How about the UWP part? I haven't found any recommendation on loading from sub-directory in UWP.

Reply:
seems like there is no "addDllDirectory" for UWP. but following the microsoft documentation it says that you need to add the path to the packagedepency in the application package manifest [https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-loadpackagedlibrary](loadpackagedlibrary)